crates/core/src/rpc/full.rs

```diff
@@ -214,25 +218,79 @@ impl Full for SurfpoolFullRpc {
         &self,
         meta: Self::Metadata,
         signature_strs: Vec<String>,
-        config: Option<RpcSignatureStatusConfig>,
+        _config: Option<RpcSignatureStatusConfig>,
     ) -> BoxFuture<Result<RpcResponse<Vec<Option<TransactionStatus>>>>> {
-        let mut statuses = vec![];
-        for sig in signature_strs.iter() {
-            statuses.push(Some(TransactionStatus {
-                slot: 0,
-                confirmations: Some(5),
-                status: Ok(()),
-                err: None,
-                confirmation_status: Some(TransactionConfirmationStatus::Finalized),
-            }));
-        }
-
-        let res = RpcResponse {
-            context: RpcResponseContext::new(0),
-            value: statuses,
+        let signatures = signature_strs
+            .iter()
+            .map(|s| Signature::from_str(&s).ok())
+            .collect::<Vec<Option<Signature>>>();
+
+        let state_reader = match meta.get_state() {
+            Ok(s) => s,
+            Err(e) => return Box::pin(future::err(e.into())),
         };
-
-        Box::pin(future::ready(Ok(res)))
+        let local_statuses = signatures
+            .iter()
+            .map(|signature| {
+                signature
+                    .map(|signature| {
+                        state_reader.svm.get_transaction(&signature).map(|tx| {
+                            tx.clone().ok().map(|_tx| TransactionStatus {
+                                slot: 0,
```

Comment:
It really feels like we should keep track of transactions processed at this layer of the stack, I'll create an issue.